src/otable/orow.py

```diff
@@ -47,12 +48,14 @@ def __getitem__(self, item: int | slice) -> T | Sequence[T]:
         if isinstance(item, slice):
             raise ValueError('slicing rows is not supported')
         assert isinstance(item, int)
-        return cast(T, getattr(self.objects[item], self.attributes[item]))
+        getter = self.columns[item].getter
+        obj = self.objects[item]
+        return cast(T, getter(obj))
 
     def __getattr__(self, name: str) -> T:
         if name in self.names:
             index = self.names.index(name)
-            return cast(T, getattr(self.objects[index], self.attributes[index]))
+            return self[index]
         raise AttributeError
```

Comment:
I generally check the error conditions first, then put the "good path" code on the normal indentation level. What I find bad here is that the `AttributeError` is removed from the condition that makes it what it is (`name not in self.names`). The function is small enough that it's not a problem here; I just wanted to mention this as a generic thing.
```python
        if name not in self.names:
            raise AttributeError(name)
        index = self.names.index(name)
        return self[index]
```